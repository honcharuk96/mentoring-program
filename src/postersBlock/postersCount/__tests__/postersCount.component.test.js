import React from 'react';
import { render } from '@testing-library/react';
import PosterCountComponent from '../postersCount.component';

describe('<PosterCountComponent />', () => {
  it('should render PosterCountComponent', () => {
    const { asFragment } = render(<PosterCountComponent numberOfPosters={0} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should contain text in  PosterCategoryComponent with 0 poster', () => {
    const { getByText } = render(<PosterCountComponent numberOfPosters={0} />);
    const text1 = getByText('0');
    const text2 = getByText('movie found');
    expect(text1);
    expect(text2);
  });

  it('should contain text in  PosterCategoryComponent with 12 posters', () => {
    const { getByText } = render(<PosterCountComponent numberOfPosters={12} />);
    const text1 = getByText('12');
    const text2 = getByText('movies found');
    expect(text1);
    expect(text2);
  });
});
