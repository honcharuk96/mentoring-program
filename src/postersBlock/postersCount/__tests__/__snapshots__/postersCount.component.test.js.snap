// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PosterCountComponent /> should render PosterCountComponent 1`] = `
<DocumentFragment>
  <div
    class="postersCountstyled__PosterCount-sc-1pea39c-0 yEgcm"
  >
    <span
      class="globalStyles__Bold-sc-osyo6d-2 SWMjf"
    >
      <span>
        0
      </span>
    </span>
     movie found
  </div>
</DocumentFragment>
`;
